src/internal/obj/limited_client.go

```diff
@@ -4,10 +4,22 @@ import (
 	"context"
 	io "io"
 	"math"
+	"time"
 
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"golang.org/x/sync/semaphore"
 )
 
+var (
+	blockedSecondsMetric = promauto.NewCounterVec(prometheus.CounterOpts{
```

Comment:
This seems like something that would make more sense as a histogram. Also, it might be good to ensure we have a consistent naming scheme for the metrics. The `object_storage` subsystem seems fine for what we will be collecting after this PR is merged, but we probably should have the subsystem either be or be prefixed by the top level subsystem (i.e. pfs / pps).

Reply:
Done.  We need a wide range here, disk is fast, the network is slow, so we have everything from ~10us to 60 seconds.  In testing, ends up looking something like this:

```
pachyderm_pfs_object_storage_limited_seconds_bucket{op="get",le="1e-05"} 8387
pachyderm_pfs_object_storage_limited_seconds_bucket{op="get",le="5e-05"} 8940
pachyderm_pfs_object_storage_limited_seconds_bucket{op="get",le="0.0001"} 8946
pachyderm_pfs_object_storage_limited_seconds_bucket{op="get",le="0.0005"} 8982
pachyderm_pfs_object_storage_limited_seconds_bucket{op="get",le="0.001"} 9008
pachyderm_pfs_object_storage_limited_seconds_bucket{op="get",le="0.005"} 9009
```

I looked to see if we could easily change the code to distinguish between "waited 10us" and "it just took 10us to get to that line of code", but the semaphore API doesn't make that particularly easy.  So I'm guessing that the smallest bucket pretty much equals "didn't block", the second smallest bucket being some noise, and the rest being meaningful.  Probably totally fine.

Reply:
Oh, also added pfs_ to the object storage-related metrics, and audited other uses of subsystem for consistency.  (The subsystems we have now are postgres, pfs, pps, pachd, and worker.  There might be some value in calling postgres pachd_postgres and worker_postgres, but it's not a super easy change to make.)